experiments/experiment.go

```diff
@@ -151,6 +210,31 @@ func (ec *experimentContext) reconcileTemplate(template v1alpha1.TemplateSpec) {
 		templateStatus.AvailableReplicas = replicasetutil.GetAvailableReplicaCountForReplicaSets([]*appsv1.ReplicaSet{rs})
 	}
 
+	// Create Service for template
+	// Use same Name and rollout-pod-template-hash as ReplicaSet
+	if template.CreateService && rs != nil {
+		podTemplateHash := rs.Labels[v1alpha1.DefaultRolloutUniqueLabelKey]
+		svc := ec.templateServices[template.Name]
+		if desiredReplicaCount == 0 {
+			ec.deleteTemplateService(svc, templateStatus, template.Name)
+		} else if svc == nil || svc.Name != rs.Name {
+			newService, err := ec.CreateService(rs.Name, template, rs.Labels)
```

Comment:
Due to the way ingress controllers work such as AWS LoadBalancer controller, Service creation needs to actually happen **_before_** pods are created, otherwise they will not get readiness gates and won't be targetted properly by the load balancer. 

So service creation needs to be moved to happen *before* we create any pods from the template.

I think the best way to handle this is to:

1. Create the ReplicaSet with `spec.replicas: 0`
2. Get PodTemplateHash of ReplicaSet (to use in Service creation)
3. Scale up ReplicaSet to desired count.

This will require some refactoring.